sklearn/gaussian_process/tests/test_tpr.py

```diff
@@ -0,0 +1,907 @@
+def test_lml_precomputed(kernel):
+    # Test that lml of optimized kernel is stored correctly.
+    tpr = TProcessRegressor(kernel=kernel).fit(X, y)
+    assert tpr.log_marginal_likelihood(tpr.kernel_.theta) == pytest.approx(
+        tpr.log_marginal_likelihood()
+    )
+
+
+@pytest.mark.parametrize("kernel", kernels)
+def test_lml_without_cloning_kernel(kernel):
+    # Test that lml of optimized kernel is stored correctly.
+    tpr = TProcessRegressor(kernel=kernel).fit(X, y)
+    input_theta = np.ones(tpr.kernel_.theta.shape, dtype=np.float64)
+
+    tpr.log_marginal_likelihood(input_theta, clone_kernel=False)
+    assert_almost_equal(tpr.kernel_.theta, input_theta, 7)
+
+
+@pytest.mark.parametrize("kernel", non_fixed_kernels)
+def test_converged_to_local_maximum(kernel):
+    # Test that we are in local maximum after hyperparameter-optimization.
+    tpr = TProcessRegressor(kernel=kernel).fit(X, y)
+
+    lml, lml_gradient = tpr.log_marginal_likelihood(tpr.kernel_.theta, True)
+
+    assert np.all(
+        (np.abs(lml_gradient) < 1e-4)
+        | (tpr.kernel_.theta == tpr.kernel_.bounds[:, 0])
+        | (tpr.kernel_.theta == tpr.kernel_.bounds[:, 1])
+    )
+
+
+@pytest.mark.parametrize("kernel", non_fixed_kernels)
+def test_solution_inside_bounds(kernel):
+    # Test that hyperparameter-optimization remains in bounds#
+    tpr = TProcessRegressor(kernel=kernel).fit(X, y)
+
+    bounds = tpr.kernel_.bounds
+    max_ = np.finfo(tpr.kernel_.theta.dtype).max
+    tiny = 1e-10
+    bounds[~np.isfinite(bounds[:, 1]), 1] = max_
+
+    assert_array_less(bounds[:, 0], tpr.kernel_.theta + tiny)
+    assert_array_less(tpr.kernel_.theta, bounds[:, 1] + tiny)
+
+
+@pytest.mark.parametrize("kernel", kernels)
+def test_lml_gradient(kernel):
+    # Compare analytic and numeric gradient of log marginal likelihood.
+    tpr = TProcessRegressor(kernel=kernel).fit(X, y)
+
+    lml, lml_gradient = tpr.log_marginal_likelihood(kernel.theta, True)
+    lml_gradient_approx = approx_fprime(
+        kernel.theta, lambda theta: tpr.log_marginal_likelihood(theta, False), 1e-10
+    )
+
+    assert_almost_equal(lml_gradient, lml_gradient_approx, 3)
+
+
+@pytest.mark.parametrize("kernel", kernels)
+def test_prior(kernel):
+    # Test that TP prior has mean 0 and identical variances.
+    tpr = TProcessRegressor(kernel=kernel)
+
+    y_mean, y_cov = tpr.predict(X, return_cov=True)
+
+    assert_almost_equal(y_mean, 0, 5)
+    if len(tpr.kernel.theta) > 1:
+        # XXX: quite hacky, works only for current kernels
+        assert_almost_equal(np.diag(y_cov), np.exp(kernel.theta[0]), 5)
+    else:
+        assert_almost_equal(np.diag(y_cov), 1, 5)
+
+
+@pytest.mark.parametrize("kernel", kernels)
+def test_sample_statistics(kernel):
+    # Test that statistics of samples drawn from TP are correct.
+    tpr = TProcessRegressor(kernel=kernel).fit(X, y)
+
+    y_mean, y_cov = tpr.predict(X2, return_cov=True)
+
+    samples = tpr.sample_y(X2, 300000)
```

Comment:
I tested this with 500 and it works so can see if we keep or leave 300000 (since scikit learn also periodically launches tests over all random seeds so better to have the simpler version)